### ISSUE TITLE
JSX occurrences request on tag does not return definition occurrence

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#17350]
From https://github.com/Microsoft/vscode/issues/31201

**TypeScript Version:**  2.4.2 and nightly

**Code**

```js
import * as React from 'react'

class Foo extends React.Component {
    render() {
        return <div></div>
    }
}

class Bar extends React.Component {
    render() {
        return <Foo></Foo>
    }
}
```

Click on `Foo` in `<Foo>` to trigger an occurrences  request

**Expected behavior:**
The  three occurrences for `Foo` are returned

**Actual behavior:**
Only the two occurrences in the jsx are returned:

```
[Trace  - 2:25:27 PM] Response received: occurrences (27). Request took 1 ms. Success: true 
Result: [
    {
        "start": {
            "line": 11,
            "offset": 17
        },
        "end": {
            "line": 11,
            "offset": 20
        },
        "file": "/Users/matb/projects/san/b.jsx",
        "isWriteAccess": false
    },
    {
        "start": {
            "line": 11,
            "offset": 23
        },
        "end": {
            "line": 11,
            "offset": 26
        },
        "file": "/Users/matb/projects/san/b.jsx",
        "isWriteAccess": false
    }
]
```

**Comment by andy-ms** [comment]
Looks like this is caused by #16453? We have special handling for open/close tags.

**Comment by andy-ms** [comment]
This was an intentional fix to #10138 which was a pretty popular issue, so we'd have to see a more feedback to consider doing this. Regular find-all-references will still work no matter where you are, it's just document highlights that changed.

**Comment by mjbvz** [comment]
Closing as VS Code no longer uses `occurrences`
